src/js/fiStatus.jsx

```diff
@@ -0,0 +1,113 @@
+var React = require('react');
+var router = require('./router.js');
+var cbLink = require('./cbLink.jsx');
+
+
+var fiStatus = React.createClass({
+
+  getInitialState: function(){
+    return this.props.institution;
+  },
+
+  getStartTime: function(editReport){
+    if(!editReport) return;
+    var startTime = editReport.timestamp;
+    var filingString = "Filing started ";
+    var timeSince = Date.now() - startTime;
+    var num;
+    var unit = ' minutes';
+
+    if(timeSince < 60000) return filingString + 'just now.';
+
+    if(timeSince >= 86000000) return filingString + 'on ' + new Date(startTime).toString().split(' ').splice(1, 3).join(' ') + '.';
+
+    if(timeSince >= 3600000){
+      unit = ' hours';
+      num = timeSince/3600000 >> 0
+    }else{
+      num = timeSince/60000 >> 0
+    }
+
+    if(num === 1) unit = unit.slice(0, -1);
+
+    return filingString + num + unit + ' ago.'
+  },
+
+  resubmitComponent: function(){
+    return <div className="resubmitComponent">{React.createElement(cbLink, {text: 'Resubmit', callback: router.resubmit})} - Resubmitting will allow you to correct any errors or invalid data encountered. Each error report will be saved so you can track your progress after resubmission.</div>
+  },
+
+  editReportsComponent: function(){
+    return (
+      <ul className="editReportsWrapper">
+        {this.state.editReports.map(function(report, i){
+          return (
+            <li key={i} className="editReport">
+              {React.createElement(cbLink, {text: "View edit report", callback: router.showErrors.bind(null, report)})} - {new Date(report.timestamp).toString().split(' ').splice(1, 3).join(' ')}
+            </li>
+          )
+        })}
+      </ul>
+    )
+  },
+
+  getStatusText: function(statusCode){
+    switch(statusCode){
+      case 0:
+        return (
+          <div className="fiStatusWrapper">
```

Comment:
We could move the
```
<div className="fiStatusWrapper">
<div className="fiStatusText">
```
and 
```
</div>
</div>
```
to the `render` function to avoid the repetition in the case statements.

Maybe to handle
```
{this.resubmitComponent()}
{this.editReportsComponent()}
```
we could check the status in `render` too?